### DISCUSSION TITLE
v2.15 |

### DISCUSSION THREAD

**Discussion by Ralim** [Ralim/IronOS#959] [General]
Improvements on the RC1 + Initial MHP30 support + Better PD version two.

As always, thank you for using this firmware 😄 

## Feature upgrades:

* MHP30 support
* Multi-lingual firmware combinations now exist for Pinecil
* More fine grained voltage controlled options
* USB-PD improvements (version one and two)
* More configuration options for power pulse
* All font / character encoding has been very reworked
* More translation updates than one can count 
* More languages 😱 

### MHP30

The MHP30 is a small reflow station from Miniware.
Thanks to a massive amount of help from @g3gg0 this firmware brings the beginnings of support for this unit.
Also kudo's to @Vinigas  and @GoJian for helping with testing.
This is not a _final_ version I'm sure, but this is a working, usable version of firmware support.
Programs the same as any one Miniware unit using drag and drop.
**Note: The boot logo scripts will need updates for this unit, so not supported yet.**

The flood doors are now open for feature requests for this unit :)

### Kudos

I have most likely missed people here, but serious thank you to all of the contributors to this firmware; and to all of the testers and users who raise issues.

@playday3008 
@t-8ch 
@HanaO00 
@xyzz 
@devydd 
@federicodilo 
@alvinhochun 
@morusn 
@TechnoLoft
@KITT3000 
@skishida 
@alvarosamudio 
@Dids 
@discip 
@hippi-viking 
@Vinigas 

<hr /><em>This discussion was created from the release <a href='https://github.com/Ralim/IronOS/releases/tag/v2.15'>v2.15 | </a>.</em>

**Comment by Purpledonut** [comment]
Hi i have a question i recently just got the TS100 and it only heats up to 440 then goes down the up then it cyckes it never goes to 450 why is that?

**Reply by vini-nu** [reply]
@Purpledonut IronOS always shows the real values.
https://github.com/Ralim/IronOS/tree/master/Documentation

Check if power supply is capable to produce enough wattage (max is 65W) on peak heating.

If problem persists, then please create github issue and output full power supply specification and output of debug-menu.
https://github.com/Ralim/IronOS/blob/master/Documentation/DebugMenu.md

Thank you!

**Reply by Ralim** [reply]
Hello,

So the tip in the ts100 is measured in relation to the cold Junction temperature, which is the inside handle temp.

It can only regulate up to around 410-420C above this temperature.

So if you set it to 450C it will go as close as it can before it runs out of measurement range which will depend on ambient temperature.

The ability to regulate at high temperatures depends on the quality and power of your power source.

**Comment by ghost** [comment]
Thanks for adopting MHP30 so quickly. Would like to see many improvements, above all reflow soldering heat profile curves.

**Reply by Ralim** [reply]
For improvements you would like to see, feel free to open an issue to discuss them.
Make sure in the issue you note down _how_ you think that should work.
For some things like reflow profiles, those can be quite tricky to manage and enter on the screen, so a bit of thought around how these should be done (do we just have pre-set ones) would be worth while.

**Reply by ghost** [reply]
Yeah, you're right. A bunch of thoughts have to go into that feature. As a first: the delta between display temp (otb firmware) and externally metered plate temp are 50 °C while in the range of 20 ... 200 °C. That's a bit of a problem tbh.

**Reply by Ralim** [reply]
Have you validated that the same temperature delta exists in this firmware?
Also did you validate it once the unit has come up to equilibrium or while it was heating?

**Comment by zhoudingbo820** [comment]
Hello, developer. I found a problem with ts100. Using this firmware will cause the display temperature to keep beating (approximately between 1°C and 5°C, I tried to calibrate the temperature but it didn't work), and it has not been able to keep within the set temperature range. There is no problem with the original firmware temperature display, I hope it can be repaired. thanks

**Reply by Ralim** [reply]
Hi,
This is a known thing with how the PID, sampling and filtering worked. This was not present on the tip of the iron itself.
This is also due to this firmware doing as little filtering as possible on the display as it leans more towards "tell the truth" than hiding things by filtering.

That said, please try a release from here as work has been done to improve this: https://github.com/Ralim/IronOS/actions/runs/1244826249

**Reply by zhoudingbo820** [reply]
> 嗨，
> 这是 PID、采样和过滤工作原理的已知问题。这在熨斗本身的尖端并不存在。
> 这也是由于这种固件在显示屏上尽可能少地进行过滤，因为它更倾向于"说实话"，而不是通过过滤来隐藏东西。
> 
> 这就是说， 请尝试从这里释放， 因为已经做了工作来改善这一点： https://github.com/Ralim/IronOS/actions/runs/1244826249

thank you very much

**Reply by zhoudingbo820** [reply]
> Hi,
> This is a known thing with how the PID, sampling and filtering worked. This was not present on the tip of the iron itself.
> This is also due to this firmware doing as little filtering as possible on the display as it leans more towards "tell the truth" than hiding things by filtering.
> 
> That said, please try a release from here as work has been done to improve this: https://github.com/Ralim/IronOS/actions/runs/1244826249

I just tried the firmware you sent and found that the jitter is much smaller than before, but there will still be a jitter of 1°C.

**Reply by Ralim** [reply]
Hi this is correct as this firmware doesn't really filter (or hide) the on screen display of the temperature at the moment.
A lot of devices out there hide small deviations but at the moment this software does not do this.

This does not affect the performance of the iron and is cosmetic only.

**Reply by discip** [reply]
Good afternoon,
I think for most of the user the present way of ***not hiding the truth*** is the desired one, even if a oscillation around **~1°C** is present, it only shows how impressively accurate this software interacts with the hardware.

@Ralim and @sandmanRO
## Thank you so much for that again! ❤👍🤝🥇

**Reply by zhoudingbo820** [reply]
> 下午好，
> 我认为对大多数用户来说，目前_**不隐藏真相**_的方法是所希望的，即使存在**1°C**左右的振荡，它也只能显示此软件与硬件交互的准确性。
> 
> @Ralim和@sandmanRO
> 
> ## 再次非常感谢！❤👍🤝🥇

I understand, thank you, but I still hope that the next firmware can freely set whether to hide the filter display temperature (because my ts100 shows that the temperature jumps too fast, I hope the displayed temperature can be more stable or slower)

**Comment by rogerlancien** [comment]
Hello developpers,

I recently updated my TS80 with your firmware, it looks great thanks for that, 
I'm having an issue where I bought a slightly bigger replacement screen, it fits but the text on the display is a little bit too big as the screen is a little bit bigger than the original one, is it possible to modify the size of the text so I can read it all ? 
thanks for answers !

**Reply by Ralim** [reply]
Sadly the answer is _not simply_.
All of the fonts are bitmapped and it would require re-creating all of the glyphs to adjust how they draw, and probably adjusting all of the screen layout code.

How different is your screen that you purchased? It may be simpler to modify the case to show the extra area if the difference is small, or just buy a suitably sized screen.

[For reference this is the font tables](https://github.com/Ralim/IronOS/blob/master/Translations/font_tables.py)

**Reply by discip** [reply]
@rogerlancien
Good evening,
did you manage to get the screen to work?
Is it an IPS screen?
If so, where did you get it from?

thanks in advance

**Reply by rogerlancien** [reply]
@discip 
Good evening, the screen is working , I didn't try to modify the size because of @Ralim 's answer, I think it's a lot of pain for a small gain, the screen is just a little bit bigger so I have to tilt it to see the entire text, not a big big deal , I just thought it would be easy to reduce the text size. cheers

**Reply by rogerlancien** [reply]
I bought it online , from german electronic seller I can't remember which one ( I think Reichelt) , it's an OLED screen

**Reply by discip** [reply]
Thank you for the almost instant answer. 😃
I am searching for an IPS, LCD ... anything that is not an OLED screen, because of its degradation.
Anyway thank you for the detailed information! 👍

kind regards

**Reply by rogerlancien** [reply]
@discip 
No problem, I started with this video https://youtu.be/yPsBWquULeM and I took a look at the datasheets and informations, maybe you can find something for you, read the description there are links in it , regards

**Reply by discip** [reply]
At the moment everything still works, but the screen got dimmer (compared to a new one).
Just wanted to be prepared, when the stock will get to dark, to be useable.

**Reply by rogerlancien** [reply]
yes that was also my initial issue, the screen turned  really dark I could barely see the text by tilting the iron

**Reply by discip** [reply]
I don't know, if you are using the latest firmware, but at some point there was a feature implemented, that allows to change the screen brightness.
I have set the lowest possible value, to prevent premature OLED failure.

**Reply by rogerlancien** [reply]
that's a great implement and a good idea to preserve the screen , I don't use mine very often tbh I mostly use bigger soldering stations, but I have to admit that those tiny irons are very useful and developers here are doing such a great job, nowadays we can hack soldering irons , that's crazy stuff
